include/toolchain/gcc.h

```diff
@@ -63,6 +63,8 @@
 #define BUILD_ASSERT(EXPR, MSG...)
 #endif
 
+#define ZRESTRICT __restrict
```

Comment:
Should use the C standard keyword when not compiling C++ code.
```suggestion
#ifdef __cplusplus
#define ZRESTRICT __restrict
#else
#define ZRESTRICT restrict
#endif
```

Reply:
The only reason to switch would be if someone is disabling GNU extensions which would most likely break other things during compilation with gcc, otherwise there is no difference in the behavior, so a single line is much cleaner than yet another conditional macro.